### ISSUE TITLE
Insufficient temporary space leads to unclear errors on backup-push

### ISSUE THREAD

**Issue by mruser** [wal-e/wal-e#4]
When there is insufficient temporary space (`/tmp`) during a `backup-push` IOError 32 (broken pipe) is raised. This issue may not be limited to `backup-push`.
#### Root cause

Insufficient `/tmp` space
#### Mitigation
1. Allocate more `/tmp` space. (May not be possible in some environments)
2. Perform `backup-push` with `--pool-size=1` to resolve higher space requirements from default of multiple workers multiplying the /tmp load.
#### Fixes
1. Document the above. `--pool-size` does not seem to be documented anywhere, I just happened to come across it while diagnosing this issue. I hope that this issue will help others encountering this problem.
2. Check for available temporary space for each worker before starting and error out gracefully.
3. Allow the user to specify a temporary path to override the default.
4. (Minor) If possible, determine if insufficient space (IOError 12) is the root cause for the broken pipe and error out gracefully.

**Comment by Uburwator** [comment]
Hi,
It looks like this issue isn't resolved, and TMPDIR workaround isn't working as well - temp files are still being created in /tmp, despite pointing TMPDIR to another location.
pool-size isn't a solution, because my /tmp partition is smaller, than default part size (1G) - is it possinle to decrease part size?

**Comment by fdr** [comment]
On Wed, Apr 13, 2016 at 6:48 PM Eugene Trufanov <notifications@github.com>
wrote:

> Hi,
> It looks like this issue isn't resolved, and TMPDIR workaround isn't
> working as well - temp files are still being created in /tmp, despite
> pointing TMPDIR to another location.
> pool-size isn't a solution, because my /tmp partition is smaller, than
> default part size (1G) - is it possinle to decrease part size?
>

Not right now. TMPDIR should work, because python's `tempfile` is used. Are
you using sudo in conjunction with wal-e, e.g. sudo -u postgres ...?
Because this can scrub your environment variables if you are not careful.

**Comment by Uburwator** [comment]
Yes, I used `sudo su postgres`, but I set up TMPDIR in /etc/wal-e/env/TMPDIR file, together with other variables, which were used in envdir command.

Fortunately I wound it works on another server, so it's not an issue with wal-e. Thank you!

**Comment by OrangeDog** [comment]
In a script run as root:
```
TMPDIR=/var/lib/postgresql/tmp
export TMPDIR

wal-e delete --confirm retain 3
wal-e backup-push "$1"
```
It still creates the files in `/tmp` and then fails due to lack of space.
